ci-operator/step-registry/ipi/conf/aws/disconnectedcluster/ipi-conf-aws-disconnectedcluster-commands.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+# subnet and AZs
+allsubnetids="${SHARED_DIR}/allsubnetids"
+availabilityzones="${SHARED_DIR}/availabilityzones"
+if [ ! -f "${allsubnetids}" ] || [ ! -f "${availabilityzones}" ]; then
+    echo "File ${allsubnetids} or ${availabilityzones} does not exist."
+    exit 1
+fi
+
+echo -e "subnets: $(cat ${allsubnetids})"
+echo -e "AZs: $(cat ${availabilityzones})"
+
+# mirror registry
+new_pull_secret="${SHARED_DIR}/new-pull-secret"
+install_config_image_patch="${SHARED_DIR}/install-config-image.yaml.patch"
+if [ ! -f "${new_pull_secret}" ] || [ ! -f "${install_config_image_patch}" ]; then
+    echo "File ${new_pull_secret} or ${install_config_image_patch} does not exist."
+    exit 1
+fi
+
+echo -e "image registry:\n$(cat ${install_config_image_patch})"
+
+public_proxy_url=$(head -n 1 "${SHARED_DIR}/public_proxy_url")
+
+CONFIG_PATCH="${SHARED_DIR}/install-config-disconnected.yaml.patch"
+cat > "${CONFIG_PATCH}" << EOF
+publish: External
+platform:
+  aws:
+    subnets: $(cat "${allsubnetids}")
+controlPlane:
+  platform:
+    aws:
+      zones: $(cat "${availabilityzones}")
+compute:
+- platform:
+    aws:
+      zones: $(cat "${availabilityzones}")
+pullSecret: >
+  $(cat "${new_pull_secret}" | jq -c .)
+additionalTrustBundle: |
+`sed 's/^/  /g' "/var/run/vault/mirror-registry/client_ca.crt"`
+proxy:
```

Comment:
I think in this case, we do not need the proxy setting

Reply:
if no proxy, two issues need to be resolved:
1) request credential via IAM endpoint, this can be resolved by Manual mode CCO, similar to disconnected&private cluster.
2) Ingress needs to create apps DNS via route53 endpoint, if no proxy, we need to do a similar process as  disconnected&private clusters: a) install failed b). create apps DNS c). `wait-for install` complete.

The proxy is also mentioned in official [document](https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-restricted-networks-aws-installer-provisioned.html#installation-configure-proxy_installing-restricted-networks-aws-installer-provisioned):

> Production environments can deny direct access to the internet and instead have an HTTP or HTTPS proxy available. You can configure a new OpenShift Container Platform cluster to use a proxy by configuring the proxy settings in the install-config.yaml file.

I think we can do the same thing as in Flexy, configure a whitelist in proxy server, to restrict cluster egress network.

Reply:
I think we can refine our existing install config in the unified ci process. I will talk with you more.